cluster/cluster.go

```diff
@@ -48,7 +48,11 @@ func (c *Cluster) AddNode(n *Node) error {
 	c.Lock()
 	defer c.Unlock()
 
-	if _, exists := c.nodes[n.ID]; exists {
+	if old, exists := c.nodes[n.ID]; exists {
+		if old.IP != n.IP {
+			log.Errorf("ID duplicated: %s", n.ID)
+			log.Errorf("IP [%s] and [%s] share the same ID", old.IP, n.IP)
```

Comment:
my own personal opinion is that this should generate a single log entry. When digging through logs, I find it unpleasant to get a matching entry, and then have to go look at surrounding lines for the complete message (`grep`, get the time, then `less` through the file to that time).

Perhaps something like:

    log.Errorf("ID duplicated. %s shared by %s and %s", n.ID, old.IP, n.IP)